datafusion/sql/src/expr/identifier.rs

```diff
@@ -47,7 +46,8 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
             // interpret names with '.' as if they were
             // compound identifiers, but this is not a compound
             // identifier. (e.g. it is "foo.bar" not foo.bar)
-            let normalize_ident = normalize_ident(id);
+            let normalize_ident =
```

Comment:
It would be really nice to refactor the code to make it harder to miss such functions. Maybe we could add a struct like

```rust
struct IdentNormalizer {
  bool normalize;
}

impl IdentNormalizer {
  fn normalize(id: Ident) -> String {
  ...
   }
}
```

And then we could change `SqlToRel` so it had a normalizer field, and remove the `fn normalize_ident` entirely -- that way the compiler could help make sure there are no more missing `normalize_ident` callsites (b/c they would all have to be changed to `self.normalizer.normalize(...)`

Reply:
I agree that a refactor here to make normalize ident a part of the SqlToRel planner itself makes more sense that constantly getting this info from the options.

Reply:
Do you have a preference for whether a new struct should be created to normalize ident or if we could just add the `normalize` method as an implementation for `SqlToRel`